app/lib/advanced_search_finder_api.rb

```diff
@@ -5,12 +5,8 @@ def content_item_with_search_results
     raise_on_missing_taxon_param
 
     filter_params[TAXON_SEARCH_FILTER] = taxon["content_id"] if taxon
-    search_response = fetch_search_response(content_item)
-    content_item_with_taxon_links = augment_content_item_links_with_taxon(
-      content_item,
-      taxon
-    )
-    augment_content_item_with_results(content_item_with_taxon_links, search_response)
+    content_item_with_results = augment_content_item_with_results
```

Comment:
Maybe either call this ```augmented_content_item``` and don't bother with the assignment, just use it directly in the statement below.

On the other hand I like explicitly including the arguments so it is clear what this methods dependencies are